src/main/java/ru/svgogin/service/spark/service/SparkService.java

```diff
@@ -43,14 +44,17 @@ public CompanyDto update(CompanyDto companyFromDb, CompanyDto companyDto) {
   }
 
   public CompanyDto save(CompanyDto companyDto) {
-    Company company = toEntity(companyDto);
-    if (sparkRepositoryDb.existsByInn(company.getInn())) {
-      throw new IllegalArgumentException("Company with Inn = "
-                                         + company.getInn()
-                                         + " already exists");
-    } else {
-      return toDto(aggregateTemplate.insert(company));
+    var companyInDb = sparkRepositoryDb.existsByInn(companyDto.getInn());
+    if (companyInDb) {
+      log.info("Company with inn "
+               + companyDto.getInn()
+               + " was not saved. Post method used instead of Put");
+      throw new EntityAlreadyExistsException("Company with inn "
+                                             + companyDto.getInn()
+                                             + " already exists");
     }
+    log.info("Company with inn " + companyDto.getInn() + " was successfully saved.");
```

Comment:
I can't swap these lines because 

log.info("Company with inn " + companyDto.getInn() + " was successfully saved.");

becomes an unreachable statement.

So this variant doesn't work.

  public CompanyDto save(CompanyDto companyDto) {
    var companyInDb = sparkRepositoryDb.existsByInn(companyDto.getInn());
    if (companyInDb) {
      log.info("Company with inn "
               + companyDto.getInn()
               + " was not saved. Post method used instead of Put");
      throw new EntityAlreadyExistsException("Company with inn "
                                             + companyDto.getInn()
                                             + " was not saved. Post method used instead of Put");
    }
    return toDto(aggregateTemplate.insert(toEntity(companyDto)));
    log.info("Company with inn " + companyDto.getInn() + " was successfully saved.");
  }

Reply:
Done. Thank you. Can you explain why this error occures?

Reply:
which error?